src/js/oTable.js

```diff
@@ -131,219 +23,346 @@ function descendingSort(...args) {
+			th.removeEventListener('keydown', this.listeners[columnIndex]);
 		});
-	});
-};
-
-/**
- *
- * @private
- * @param {Number} columnIndex
- */
-OTable.prototype._sortByColumn = function _sortByColumn(columnIndex) {
-	return function (event) {
-		const currentSort = event.currentTarget.getAttribute('aria-sort');
-		const sort = this.rootEl.getAttribute('data-o-table-order') === null || currentSort === "none" || currentSort === "descending" ? 'ASC' : 'DES';
-
-		/**
-		 * Check if sorting has been cancelled on this table in favour of a custom implementation.
-		 *
-		 * The return value is false if event is cancelable and at least one of the event handlers
-		 * which handled this event called Event.preventDefault(). Otherwise it returns true.
-		 * https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent
-		 */
-		const customSort = !event.currentTarget.dispatchEvent(new CustomEvent('oTable.sorting', {
-			detail: {
-				sort,
-				columnIndex,
-				oTable: this
-			},
-			bubbles: true,
-			cancelable: true
-		}));
+	}
 
-		if (!customSort) {
-			this.sortRowsByColumn(columnIndex, sort === "ASC", event.currentTarget.getAttribute('data-o-table-data-type') === 'numeric');
+	/**
+	 * Sorts the table by a specific column
+	 * @param {number} index The index of the column to sort the table by
+	 * @param {bool} sortAscending Which direction to sort in, ascending or descending
+	 * @param {bool} isNumericValue Deprecated: Set `type` instead.
+	 * @param {string} type What type of data the column holds to enable sorting of numeric values, dates, etc.
+	 * @returns undefined
+	 */
+	sortRowsByColumn(index, sortAscending, isNumericValue = null, type = null) {
+		if (isNumericValue !== null) {
+			console.warn(`"sortRowsByColumn" argument "isNumericValue" is deprecated. Set "type" to a valid type such as "numeric" or "text". More specific types are listed in the README https://github.com/Financial-Times/o-table#sorting.`);
 		}
+		// If type is not set but deprecated "isNumericValue" is, set the type to numeric.
+		if (isNumericValue) {
+			type = type || 'numeric';
+		}
+		const tbody = this.rootEl.querySelector('tbody');
+		const rows = Array.from(tbody.querySelectorAll('tr'));
+		const intlCollator = getIntlCollator();
+		rows.sort(function (a, b) {
+			let aCol = a.children[index];
+			let bCol = b.children[index];
+			aCol = tableSortFormatter.formatCell({ cell: aCol, type });
+			bCol = tableSortFormatter.formatCell({ cell: bCol, type });
+			if (sortAscending) {
+				return ascendingSort(aCol, bCol, intlCollator);
+			} else {
+				return descendingSort(aCol, bCol, intlCollator);
+			}
+		});
 
-		/**
-		 * Update aria attributes to provide immediate feedback.
-		 *
-		 * This is called again by the `sorted` method to assure accuracy.
-		 * I.e. if a sort fails previous sort attributes can be restored via the `sorted` method.
-		 */
+		window.requestAnimationFrame(() => {
```

Comment:
Need to include this in the browser features in origami.json

Reply:
Looks like `requestAnimationFrame` is supported in all browsers we support an enhanced experience.